pkg/registry/flowcontrol/prioritylevelconfiguration/strategy.go

```diff
@@ -84,7 +87,24 @@ func (priorityLevelConfigurationStrategy) PrepareForUpdate(ctx context.Context,
 
 // Validate validates a new priority-level.
 func (priorityLevelConfigurationStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
-	return validation.ValidatePriorityLevelConfiguration(obj.(*flowcontrol.PriorityLevelConfiguration), getRequestGroupVersion(ctx))
+	newPL := obj.(*flowcontrol.PriorityLevelConfiguration)
+	gvk := getRequestGroupVersion(ctx)
+
+	// 1.28 server is not aware of the roundtrip annotation, and will
+	// default any 0 value persisted (for the NominalConcurrencyShares
+	// field of a priority level configuration object) back to 30 when
+	// reading from etcd.
+	// That means we should not allow 0 values to be introduced, either
+	// via v1 or v1beta3(with the roundtrip annotation) until we know
+	// all servers are at 1.29+ and will honor the zero value correctly.
+	//
+	// TODO(121510): 1.29: don't allow a zero value, either via v1 or
+	//  v1beta3 (with the roundtrip annotation) for the
+	//  'nominalConcurrencyShares' field of 'limited' for CREATE operation.
+	//  1:30: lift this restriction, allow zero value via v1 or v1beta3
+	allErrs := validation.ValidateCreatePriorityLevelConfiguration(newPL, gvk)
+	allErrs = append(allErrs, validation.ValidatePriorityLevelConfiguration(newPL, gvk)...)
```

Comment:
see comments in the validation file, I think we want to keep a single function and pass in options about whether to allow zero or not